subprojects/studio-plugin/src/main/java/org/gradle/profiler/studio/plugin/system/AndroidStudioSystemHelper.java

```diff
@@ -1,32 +1,40 @@
 package org.gradle.profiler.studio.plugin.system;
 
-import com.android.tools.idea.gradle.project.sync.GradleSyncInvoker;
 import com.android.tools.idea.gradle.project.sync.GradleSyncState;
+import com.android.tools.idea.projectsystem.ProjectSystemSyncManager;
+import com.android.tools.idea.projectsystem.ProjectSystemUtil;
 import com.intellij.openapi.application.ApplicationManager;
 import com.intellij.openapi.progress.ProgressIndicator;
 import com.intellij.openapi.project.Project;
+import com.intellij.openapi.startup.StartupManager;
 import com.intellij.openapi.wm.IdeFrame;
 import com.intellij.openapi.wm.ex.StatusBarEx;
 import com.intellij.openapi.wm.ex.WindowManagerEx;
 import com.intellij.util.messages.MessageBusConnection;
 import org.gradle.profiler.studio.plugin.system.GradleProfilerGradleSyncListener.GradleSyncResult;
 
-import static com.google.wireless.android.sdk.stats.GradleSyncStats.Trigger.TRIGGER_USER_SYNC_ACTION;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Supplier;
+
+import static com.android.tools.idea.projectsystem.ProjectSystemSyncUtil.PROJECT_SYSTEM_SYNC_TOPIC;
 
 public class AndroidStudioSystemHelper {
 
     private static final long WAIT_ON_PROCESS_SLEEP_TIME = 10;
+    private static final long WAIT_ON_STARTUP_SLEEP_TIME = 100;
 
     /**
      * Does a Gradle sync.
      */
     public static GradleSyncResult doGradleSync(Project project) {
-        GradleProfilerGradleSyncListener syncListener = new GradleProfilerGradleSyncListener();
+        GradleProfilerGradleSyncListener syncListener = subscribeToGradleSync(project);
         try {
-            GradleSyncInvoker.getInstance().requestProjectSync(project, TRIGGER_USER_SYNC_ACTION, syncListener);
-        } catch (Exception e) {
-            e.printStackTrace();
-            syncListener.syncFailed(project, e.getMessage());
+            // We could get sync result from the `Future` returned by the syncProject(),
+            // but it doesn't return error message so we rather listen to GRADLE_SYNC_TOPIC to get the sync result
+            ProjectSystemUtil.getSyncManager(project).syncProject(ProjectSystemSyncManager.SyncReason.USER_REQUEST).get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new RuntimeException(e);
```

Comment:
Should we still call `synclistener.syncFailed()` here, or is it okay to just throw?

Reply:
Maybe it's better to call sync failed, to not crash our plugin.

Reply:
We are now back to throwing an exception instead of `syncListener.failed()`, is this intentional?

Reply:
Hmm, it seems github is showing outdated version here. Current version is here:
https://github.com/gradle/gradle-profiler/pull/439/files#diff-8eb36c0b3730fb2241bb7c46b8367c29778fcf7971216d69e1fb5df5d9938d1e